index.bs

```diff
@@ -3209,6 +3210,81 @@ opened</dfn> steps given |window|, |type| and |message|.
 
 </div>
 
+## Emulation ## {#module-emulation}
+
+The <dfn export for=modules>emulation</dfn> module contains functionality for
+emulating device-dependent and system-dependent parameters.
+
+### Definition ### {#module-emulation-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+EmulationCommand = (
+  emulation.SetViewport
+)
+</pre>
+
+### Types ### {#module-emulation-types}
+
+### Commands ### {#module-emulation-commands}
+
+#### The emulation.setViewport Command ####  {#command-emulation-setViewport}
+
+The <dfn export for=commands>emulation.setViewport</dfn> command emulates specific viewport characteristics (e.g. viewport width and viewport height) on the given top-level browsing context.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      emulation.SetViewport = {
+        method: "emulation.setViewport",
+        params: emulation.SetViewportParameters
+      };
+
+      emulation.SetViewportParameters = {
+        context: browsingContext.BrowsingContext,
+        ? viewport: emulation.Viewport
```

Comment:
I don't love making it optional and using the missing value default as "reset", especially given that command parameters are general assumed to be Extensible, and so sending a payload like `{"veiwport: {"width": 800, "height": 600}}` would silently reset the override. I don't know if we have existing precedent in the spec, but using an explicit value like `null` seems preferable.

Reply:
Updated.